### PR TITLE
Homework3: Alexander Shvedov

### PR DESCRIPTION
**Pull request by alexshvedov1997**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle).
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks.
- [ ] my PR made from the fork feature branch to appropriate branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+for elem in range(1, 101):
+    if not elem % 3 and not elem % 5:
```
Comment (belovmd) [inline src/homework3/task1.py:9]: 15?

File: src/homework3/task2.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+lst_ = [chr1 + chr2 for chr1 in "ab" for chr2 in "bcd"]
+print(*lst_)
+print(*lst_[::2])
+lst_2 = [str(chr1) + chr2 for chr1 in range(1, 5) for chr2 in "a"]
```
Comment (belovmd) [inline src/homework3/task2.py]: '1234'

Comment (belovmd) [reply src/homework3/task2.py]: +'a'

File: src/homework3/task3.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+set_1 = set(['a', 'b', 'c'])
+print("set: {} type: {}".format(set_1, type(set_1)))
```
Comment (belovmd) [inline src/homework3/task3.py]: redundant print

File: src/homework3/task4.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import random
+
+random.seed(4400)
+
+
+def calculate_count_of_pairs(count_num):
+    count_of_pairs = 0
+    while count_num:
```
Comment (belovmd) [inline src/homework3/task4.py]: has O(1) solution

**Review by belovmd — CHANGES_REQUESTED** [review]
sanity checks failed - no task description

File: src/homework3/task4.py (hunk 4 above)
Comment (alexshvedov1997) [reply src/homework3/task4.py]: Я оптимизировал код, но не знаю как еще улучшить код для списка произвольных чисел.

File: src/homework3/task1.py (hunk 1 above)
Comment (alexshvedov1997) [reply src/homework3/task1.py:9]: I don't understand what's wrong with the programm.

Comment (belovmd) [reply src/homework3/task1.py:9]: `if not element %15:`

review not addressed

File: src/homework3/task2.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+4.Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5.Скопируйте список и добавьте в него элемент '2a' так чтобы в
+исходном списке этого элемента не было.
+'''
+
+
+lst_ = [chr1 + chr2 for chr1 in "ab" for chr2 in "bcd"]
+print(*lst_)
+print(*lst_[::2])
+lst_2 = [str(chr1) + 'a' for chr1 in range(1, 5)]
```
Comment (belovmd) [inline src/homework3/task2.py:18]: chr1 in '1234' review not addressed

File: src/homework3/task3.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+3.Используйте генератор списков чтобы получить следующий
+['1a', '2a', '3a', '4a'].
+4.Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5.Скопируйте список и добавьте в него элемент '2a' так чтобы в
+исходном списке этого элемента не было.
+'''
+
+
+set_1 = set(['a', 'b', 'c'])
+print("set: {}".format(set_1))
```
Comment (belovmd) [inline src/homework3/task3.py:16]: redundant print. review not addressed

File: src/homework3/task3.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+'''
+List practice
```
Comment (belovmd) [inline src/homework3/task3.py:2]: Task description contradicts solution. (it's description from task2)

File: src/homework3/task4.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+'''
+
+
+import random
+
+random.seed(4400)
+
+
+def calculate_count_of_pairs(lst_):
```
Comment (belovmd) [inline src/homework3/task4.py:16]: here we have arithmetical progression. number of pairs - n(n-1)/2

**Review by belovmd — APPROVED** [review]
7,5

**Pull request closed without merging**